server/datastore/s3/installer.go

```diff
@@ -0,0 +1,62 @@
+package s3
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+)
+
+// InstallerParams describes the params used to locate an installer in an
+// S3 bucket
+type InstallerParams struct {
```

Comment:
Missing prefix string.

Reply:
that one is set as part of the S3 config

Reply:
It's set when the store is initialized here:

https://github.com/fleetdm/fleet/blob/933420777932cc7a8d4d63987de47f0d87e63228/server/datastore/s3/s3.go#L68-L72

But agreed that `Bucket` is set there too, so either we remove both or we over-specify both to be consistent 👍  will update

Reply:
I had to revert this change, turns out `Bucket` is required in the S3 calls, AND you can't pass a prefix.